src/ui/api-helper.js

```diff
@@ -0,0 +1,77 @@
+import { useMemo } from 'react';
+import { ERROR_CODES } from '../domain/constants.js';
+import { useDialogs } from '../components/dialog-context.js';
+import { useService } from '../components/container-context.js';
+
+export async function tryApiCallWithLoginFallback({ executeCall, onLoginRequested }) {
+  let result = null;
+
+  let firstTryError = null;
+  try {
+    result = await executeCall();
+  } catch (err) {
+    firstTryError = err;
+  }
+
+  if (!firstTryError) {
+    return result;
+  }
```

Comment:
Actually this `if` can go away and just return on line 11, right? Would simplify this long method a bit.
I guess you wanted to make everything more explicit, also with the `secondTryError`, but I would say we could just replace lines `31` -> `42` with `return executeCall()` ?

Reply:
Thanks for the suggestion, it looks a lot simpler now.